### PR TITLE
libwebsockets and mosquitto: Prepare for PR #3822

### PR DESCRIPTION
**Pull request by slewsys**
#### Description
mosquitto and libwebsockets: Updates for PR #3822 

Testing upgrade to OpenSSL 1.1 raised two issues:
- Building __net/mosquitto__ failed because PR #3822 does not revbump __devel/libwebsockets__,
- Testing  __net/mosquitto__ failed if mosquitto was not already installed.

This PR addresses both problems.  A new revision is omitted for __net/mosquitto__ so as to not interfere with PR #3822.

###### Tested on

macOS 10.12.6 16G1815
Xcode 9.2 9C40b 

###### Verification 
Have you

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/mosquitto/Portfile [hunk 1]
```diff
@@ -51,8 +51,8 @@ pre-build {
 
 pre-test {
     if {![file exist ${build.dir}/test]} {
-        foreach path [list config.mk test] {
-            copy -force ${worksrcpath}/${path} ${build.dir}
+        foreach path [list config.mk test lib/mosquitto.h lib/cpp/mosquittopp.h src/mosquitto_broker.h src/mosquitto_plugin.h] {
+          copy -force ${worksrcpath}/${path} ${build.dir}/${path}
```
[the remark this answers is not in the document]
Comment (slewsys) [reply net/mosquitto/Portfile]: Oops!  Corrected, thank you!

**Comment by pmetzger** [comment]
There are conflicts. @mojca can you steer this to completion?

**Comment by yan12125** [comment]
> Testing net/mosquitto failed if mosquitto was not already installed.

I think this is also an issue with openssl 1.0.2?

> Building net/mosquitto failed because PR #3822 does not revbump devel/libwebsockets,

FYI, this is now fixed in #3822

**Comment by slewsys** [comment]
> > Testing net/mosquitto failed if mosquitto was not already installed.
> 
> I think this is also an issue with openssl 1.0.2?

Correct.

> > Building net/mosquitto failed because PR #3822 does not revbump devel/libwebsockets,
> 
> FYI, this is now fixed in #3822

Okay.  Would this PR be better resubmitted after PR #3822 is merged?
In the mean time, I'll remove the libwebsockets revbump.

**Comment by slewsys** [comment]
> There are conflicts. @mojca can you steer this to completion?

My local branch was outdated.  It is now up to date.  Sorry!

**Comment by yan12125** [comment]
LGTM if commits are squashed.

@mojca?

**Comment by yan12125** [comment]
Seems mojca is busy on other ports/projects. As her comment is already addressed, I'm going to merge this.

**Comment by yan12125** [comment]
(I use "squash and merge" and leave relevant commit messages only. @slewsys Please squash your commits locally and force-push instead of introducing merge commits the next time.)

**Comment by slewsys** [comment]
On Mar 18, 2019, at 9:26 AM, Chih-Hsuan Yen <notifications@github.com> wrote:
> 
> (I use "squash and merge" and leave relevant commit messages only. @slewsys Please squash your commits locally and force-push instead of introducing merge commits the next time.)

Thank you for the heads up and apologies!  `Rebasing against' master rather than `merging with' is my usual workflow.  Will pay more attention to unintended side-effects!
-AM

**Pull request merged**